dvc/repo/plots/__init__.py

```diff
@@ -84,7 +84,8 @@ def show(self, targets=None, revs=None, props=None, templates=None):
         # If any mentioned plot doesn't have any data then that's an error
         targets = [targets] if isinstance(targets, str) else targets or []
         for target in targets:
-            if not any("data" in d[target] for d in data.values()):
+            rpath = relpath(target, self.repo.root_dir)
+            if not any("data" in d[rpath] for d in data.values()):
```

Comment:
`collect` returns data in as a mapping: `revision` -> `relative_path` -> `data` - so any time one would try to run `dvc plots diff/show` from other cwd that repo root, the data would not be found.